server/src/main/java/org/elasticsearch/index/codec/vectors/es910/hnsw/HnswUtil.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * @notice
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Modifications copyright (C) 2025 Elasticsearch B.V.
+ */
+package org.elasticsearch.index.codec.vectors.es910.hnsw;
+
+import org.apache.lucene.internal.hppc.IntHashSet;
+import org.apache.lucene.util.FixedBitSet;
+import org.apache.lucene.util.hnsw.HnswGraph;
+
+import java.io.IOException;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.List;
+
+import static org.apache.lucene.search.DocIdSetIterator.NO_MORE_DOCS;
+
+/** Utilities for use in tests involving HNSW graphs */
+public class HnswUtil {
+
+    // utility class; only has static methods
+    private HnswUtil() {}
+
+    // Finds orphaned components on the graph level.
+    static List<Component> components(HnswGraph hnsw, int level, FixedBitSet notFullyConnected, int maxConn) throws IOException {
```

Comment:
I don't think any of this is used now, maybe we can remove it.

Reply:
That's correct - removed in 457e9a462ae3ee93df9d3496ab624a534eff27bb